cmd/stoic/main.go

```diff
@@ -3,22 +3,24 @@ package main
 import (
 	"os"
 	"path/filepath"
+	"strings"
 
 	"github.com/stoic-cli/stoic-cli-core/cmd/stoic/cmd"
 )
 
 func translateSymlink() {
-	executable, err := os.Executable()
-	if err != nil {
-		return
+	_, tool := filepath.Split(os.Args[0])
+	if strings.HasPrefix(tool, "stoic") {
+		if len(tool) == len("stoic") ||
+			strings.IndexByte("-.0123456789_", tool[len("stoic")]) != -1 {
```

Comment:
What is it intended for these bytes to match?

Reply:
The main point of this change was to still treat suffixed names, e.g. `stoic-1.0.0-darwin-amd64`, as `stoic`, instead of trying to find a tool with that name.

If the tool name starts with `stoic`, this then checks that the character that follows is a number, `-`, `.`, or `_`. This lets us take over all things `stoic`, without taking over, say, `stoicism`.

There probably aren't that many words that would clash with a full takeover of the `stoic` prefix, but this is just trying to be reasonably conservative